### PR TITLE
Avoid send empty batches for Hash partitioning.

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?
#1405 
Closes #.

### CODE REVIEW

File: datafusion/src/physical_plan/repartition.rs [hunk 1]
```diff
@@ -326,6 +326,11 @@ impl RepartitionExec {
                 Partitioning::Hash(exprs, _) => {
                     let timer = r_metrics.repart_time.timer();
                     let input_batch = result?;
+                    //avoid send empty batch to next plan
+                    if input_batch.num_rows() == 0 {
```
[the remark this answers is not in the document]
Comment (Ted-Jiang) [reply datafusion/src/physical_plan/repartition.rs]: it was my mistake. Thanks a lot for your advice!  It's more reasonable for add check in the generate side 😂

[the remark this answers is not in the document]
Comment (Ted-Jiang) [reply datafusion/src/physical_plan/repartition.rs]: it was my mistake. Thanks a lot for your advice!  It's more reasonable for add check in the generate side 😂

File: datafusion/src/physical_plan/repartition.rs [hunk 2]
```diff
@@ -348,6 +348,9 @@ impl RepartitionExec {
                     for (num_output_partition, partition_indices) in
                         indices.into_iter().enumerate()
                     {
+                        if partition_indices.is_empty() {
```
Comment (Dandandan) [inline datafusion/src/physical_plan/repartition.rs:351]: 👍perfect!

**Review by Dandandan — APPROVED** [review]

**Comment by Ted-Jiang** [comment]
fix UT check in #1462

**Comment by Ted-Jiang** [comment]
retest this please

**Comment by Dandandan** [comment]
> fix UT check in #1462 

Could you rebase to fix the linting errors?

**Comment by Ted-Jiang** [comment]
it seems no CI error, i rebase -i upstream/master

> > fix UT check in #1462
> 
> Could you rebase to fix the linting errors?

it seems no CI error, i rebase -i upstream/master and force push it

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thank you @Ted-Jiang !

**Pull request merged**